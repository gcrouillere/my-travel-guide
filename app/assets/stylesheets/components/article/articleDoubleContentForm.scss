.article-container-form {
  .double-content-elements {
    display: flex;
    justify-content: center;
    border: 1px solid #ccc;
    border-radius: 2px;
    .photoInput, .mapInput, .textContentInput {
      margin: 0;
      width: 50%;
    }
    .photoInput {
      .mixedPhotoMobile {
        display: none;
      }
    }
    .mapInput {
      .mapBloc {
        height: 100%;
        display: flex;
        flex-direction: column;
        .googleMap {
          height: auto;
          flex: 1 0 auto;
        }
      }
    }
    .textContentInput {
      display: flex;
      flex-direction: column;
      .quill {
        height: 100%;
        display: flex;
        flex-direction: column;
        .ql-toolbar.ql-snow {
          height: auto;
        }
        .ql-container.ql-snow {
          height: auto;
          flex: 1 0 auto;
        }
      }
    }
  }
  .double-content-elements > div:nth-of-type(2) {
    border-right: 1px solid #ccc;
  }

  @media all and (max-width:500px) {
    .elementResize {
      display: none;
    }
    .double-content-elements {
      display: block;
      height: auto !important;
      .photoInput, .mapInput, .textContentInput {
          width: 100%;
      }
      .photoInput {
        background-image: none !important;
        .mixedPhotoMobile {
         display: block;
        }
      }
    }
  }
}
