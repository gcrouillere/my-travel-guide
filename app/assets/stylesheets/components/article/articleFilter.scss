.articlesFilter .audienceFilter {
  .form-check {
    label {
      padding: 5px;
    }
  }
}

.audienceFilter.audienceSelection {
  margin: 10px auto;
}

.stringFilter {
  position: relative;
  width: 52%;
  @media all and (max-width:768px) {
    width: 82%;
  }
  margin: 10px auto 30px auto !important;
  .stringFilterInput {
    padding: 5px;
    outline: none;
    border-radius: 30px;
    width: 100%;
    text-align: center;
    border: 1px solid #ccc;
    text-shadow: 0px 1px 0px rgba(0,0,0,0.2);
    background: white;
  }
}

.stringFilter.ticked span:after {
  position: absolute;
  top: -4px; bottom: -4px;
  left: -4px; right: -4px;
  background: linear-gradient(to right, #343a40, #00bf8f);
  opacity: 1;
  content: '';
  z-index: -1;
  border-radius: 30px;
  transition: all 0.3s ease-in-out;
}
.stringFilter.blank span:after {
  position: absolute;
  top: -4px; bottom: -4px;
  left: -4px; right: -4px;
  background: linear-gradient(to right, #343a40, #00bf8f);
  opacity: 0;
  content: '';
  z-index: -1;
  border-radius: 30px;
  transition: all 0.3s ease-in-out;
}

.filters {
  height: auto;
  display: flex;
  flex-direction: column;
  position: relative;
  .mapRow {
    height: 100%;
  }
  .mapRow.hide {
    display: none;
  }
  .googleMap {
    height: 100%;
    width: 100%;
  }
  .listSwitch {
    position: fixed;
    display: flex;
    flex-direction: column;
    z-index: 1;
    .switchView {
      margin: 8px;
      cursor: pointer;
      padding: 1px;
      width: 33px;
      height: 33px;
      border-radius: 50%;
      background: white;
      color: white;
      box-shadow: 0 0 20px rgba(0,0,0,.2);
      text-shadow: 1px 1px 1px rgba(0,0,0,0.6);
      border: 2px solid white;
    }
  }
}
.filters.full {
  height: calc(100vh - 76px);
  padding-bottom: 20px;
}
.listOnMap {
  width: 100%
}
