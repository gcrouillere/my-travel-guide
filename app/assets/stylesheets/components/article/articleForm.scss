.article-container {
  padding-right: 30px !important;
  padding-left: 30px !important;

  .audienceSelection {
    margin: 30px 0;
  }

  .blocAddition {
    margin: 30px 0;
  }

  header {
    margin: 30px 0;
    input {
      width: 100%;
      height: 28px;
      border-radius: 2px;
      padding: 25px;
      border: 1px solid #ccc;
      transition: .3s;
      box-shadow: 1px 1px 0 rgba(241,241,241,0.35);
      font-size: 25px;
    }
  }

  .textContentInput {
    margin: 30px 0;
    .ql-container {
      border-radius: 0px 0px 2px 2px;

      .ql-toolbar {
        border-radius: 2px 2px 0px 0px;
      }

     .ql-editor {
        font-family: inherit;
        line-height: 22px;
      }
    }
  }

}
