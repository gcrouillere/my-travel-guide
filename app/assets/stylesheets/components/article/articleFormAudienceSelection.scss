.audienceSelection {
    transition: all 0.3s ease-in-out;
    margin: 30px 0;

    .form-check {
      display: flex;
      justify-content: center;
      align-items: center;
      margin: 10px auto 30px auto;
      input {
        display: none;
      }
      label {
        cursor: pointer;
        margin: 0 10px;
        border-radius: 30px;
        padding: 10px;
        width: 130px;
        text-align: center;
        border: 1px solid #ccc;
        text-shadow: 0px 1px 0px rgba(0,0,0,0.2);
        background: white;
        position: relative;
      }
      label.ticked:after {
        position: absolute;
        top: -4px; bottom: -4px;
        left: -4px; right: -4px;
        background: linear-gradient(to right, #343a40, #00bf8f);
        opacity: 1;
        content: '';
        z-index: -1;
        border-radius: 30px;
        transition: all 0.3s ease-in-out;
      }
      label.blank:after {
        position: absolute;
        top: -4px; bottom: -4px;
        left: -4px; right: -4px;
        background: linear-gradient(to right, #343a40, #00bf8f);
        opacity: 0;
        content: '';
        z-index: -1;
        border-radius: 30px;
        transition: all 0.3s ease-in-out;
      }
    }

    .AudienceSubmitComplete {
      opacity: 1;
      transition: all 0.3s ease-in-out;
    }
    .AudienceSubmit{
      opacity: 0;
    }
    .AudienceSubmitContinueAndComplete {
      display: none;
    }
  }

  .audienceSelection.incomplete {
    overflow-y: auto;
    position: fixed;
    top: 0;
    height: 100vh;
    left: 0;
    width: 100%;
    transition: all 0.3s ease-in-out;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    background: #fafafa;
  }
