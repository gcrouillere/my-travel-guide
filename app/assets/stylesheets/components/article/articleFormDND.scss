.dropZone-before, .dropZone-after {
  width: 100%;
  border: none;
  opacity: 0;
  height: 0px;
  p {
    height: 0;
    margin: 0;
  }
}

.dropZone-before.active, .dropZone-after.active {
  opacity: 1;
  height: auto;
  margin: 30px 0;
  padding: 10px;
  border: 1px dashed #ccc;
  transition: all 0.3s ease-in-out;
  animation: dropZoneBlink 1.3s ease-in-out infinite alternate;
  p {
    height: auto;
    margin-bottom: 11px;
    padding: 20px;
    margin: 0;
    animation: dropHighlight 1.3s ease-in-out infinite alternate;
  }
}

#dragImage {
  width: 70%;
  border: 1px dashed #ccc;
  padding: 10px;
  position: absolute;
  top: - 200px;
 p {
    padding: 20px;
    margin: 0;
    background-color: #F7F7F7;
  }
  .imageDragging {
      display: flex;
      justify-content: center;
      align-items: center;
    img {
      width: 20%;
    }
  }
}

.dragOverlay {
  opacity: 0;
  position: absolute;
  z-index: -1;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  transition: all 0.3s ease-in-out;
  p {
    margin: 0;
    font-size: 20px;
    letter-spacing: 2px;
    color: white;
    text-shadow: 0px 2px 0px rgba(0,0,0,0.15);
  }
}

.draggingElement .dragOverlay {
  z-index: 2;
  opacity: 0.8;
  background: #343a40;
  transition: all 0.3s ease-in-out;
}

@keyframes dropHighlight {
  from {
    background: #fff;
  }

  to {
    background: #F7F7F7;
  }
}

@keyframes dropZoneBlink {
  from {
    border: 1px dashed #ccc;
  }

  to {
    border: 1px solid #ccc;
  }
}
