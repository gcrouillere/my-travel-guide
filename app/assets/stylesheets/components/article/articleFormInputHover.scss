.textContentInput, .mapInput {
  position: relative;
  .contentDelete, .dragLogo, .resizeLogo {
    position: absolute;
    background: #343a40;
    color: white;
    box-shadow: 0px 0px 0px 3px #343a40;
    padding: 1px;
    width: 20px;
    height: 20px;
    border: none;
    opacity: 0;
    img {
      display: block;
      width: 100%;
    }
  }
  .contentDelete {
    top: -24px;
    right: 0px;
    cursor: pointer;
  }
  .dragLogo {
    top: -24px;
    left: 0px;
    cursor: move;
  }
  .resizeLogo {
    bottom: -24px;
    left: calc(50% - 12px);
    cursor: ns-resize;
  }
}

.textContentInput:not(.dragging) , .mapInput:not(.dragging) {
  &:hover {
    box-shadow: 0px 0px 0px 3px #343a40;
    border-radius: 0 0 1px 1px;
    cursor: move;
    .dragLogo, .contentDelete, .resizeLogo {
      opacity: 1;
    }
    .mapBloc {
      border-radius: 0px;
    }
    .ql-toolbar {
      border-radius: 0px;
    }
    .ql-container {
      border-radius: 0px;
    }
  }
}
