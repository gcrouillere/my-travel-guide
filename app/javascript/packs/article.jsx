import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import $ from 'jquery'
import update from 'immutability-helper'

import ajaxHelpers from './../utils/ajaxHelpers'
import mapHelper from './../utils/mapHelper'
import photoHelpers from './../utils/photoHelpers'
import orderHelper from './../utils/articleContentHelper'
import markerLogos from './articleForm/mapForm/markerLogos'
import ShowSecondContentButton from './articleForm/formElementManagement/showSecondContentButton'

export class Article extends Component {
  constructor(props) {
    super(props)
    this.state = {
      articleElements: [],
      maps: [],
      texts: [],
      title: "",
      clientWidth: document.body.clientWidth,
      doubleContentActives: {}
    }
    this.textContentDivs = {}
    this.mapDivs = {}
    this.articleContent = React.createRef()
  }

  componentDidUpdate(prevProps) {
    if (this.checkUpdate(prevProps, this.props)) {
      this.setState({
        articleElements: orderHelper.orderArticleElements(this.props.currentArticle),
        maps: this.concatContentList("maps"),
        texts: this.concatContentList("text_contents"),
        title: this.props.currentArticle.title,
        doubleContentActives: this.doubleContentInitialStates()
      }, () => {
        this.createMaps()
        this.createTextContentsHTML()
      })
    }
  }

  doubleContentInitialStates = () => {
    let doubleContentActives = this.state.doubleContentActives
    this.props.currentArticle.double_contents.forEach( x => {
      const key = `doubleContent-${x.id}`
      doubleContentActives = update(doubleContentActives, {[key]: { $set: false }})
    })
    return doubleContentActives
  }

  checkUpdate(prevProps, newProps) {
    if (!prevProps.currentArticle && newProps) return true
    if (prevProps.currentArticle) {
      if (prevProps.currentArticle.id !== newProps.currentArticle.id) return true
    }
    if (newProps.currentArticle.title &&
        newProps.currentArticle.title != "" &&
        this.articleContent.current.innerText === "") return true
    return false
  }

  concatContentList = (content) => {
    let articleContent = this.props.currentArticle[content]
    let doubleContentsContent = this.props.currentArticle.double_contents.reduce((acc, x) => acc.concat(x[content]), [])
    return articleContent.concat(doubleContentsContent)
  }

  createMaps() {
    Object.keys(this.mapDivs).map(mapID => {
      const map = this.state.maps.find( x => x.id === parseInt(mapID))
      const googleMap = this.createMap(map, mapID)

      let markers = map.markers ||Â []
      this.addCenter(map, markers)
      markers.forEach( marker => {

        const icon = markerLogos[marker.logo].url
        const googleMarker = mapHelper.createMarker(marker, googleMap, icon, null, false)
        if (marker.description) {
          const infowindow = new google.maps.InfoWindow({content: marker.description, disableAutoPan: true});
          infowindow.open(googleMap, googleMarker);
        }
      })

      const polylines = map.polylines || []
      polylines.forEach( polyline => {

       const googlePolyline = this.createPolyline(polyline, googleMap)
       polyline.markers.forEach((marker, index) => {

        let icon = mapHelper.getPolylinePointLogo(index, polyline.markers)
        let tempMarker = mapHelper.createMarker(marker, googleMap, icon, index)
        mapHelper.manageDistanceInfoWindow(index, polyline.distance_displayed, polyline.markers,
        googlePolyline, tempMarker, googleMap)
       })
      })
    })
  }

  createMap(map, mapID) {
    return new google.maps.Map( this.mapDivs[mapID], {
      center: { lat: map.lat, lng: map.lng },
      zoom: map.zoom,
      mapTypeControl: true,
      zoomControl: true,
      zoomControlOptions: { position: google.maps.ControlPosition.LEFT_CENTER },
      scaleControl: false,
      streetViewControl: false,
      fullscreenControl: false,
      styles: [ { "featureType": "poi.business", "stylers": [{ "visibility": "off" }] } ]
    });
  }

  createPolyline(polyline, googleMap) {
    return new google.maps.Polyline({
      map: googleMap,
      path: polyline.markers,
      strokeColor: '#495057',
      strokeOpacity: 1.0,
      strokeWeight: 2,
      geodesic: true
    })
  }

  addCenter(map, markers) {
    if (map.show_map_center_as_marker) markers.unshift({
      lat: map.lat,
      lng: map.lng,
      description: map.name,
      logo: "markerLogo",
      mapCenter: true
    })
  }

  createTextContentsHTML() {
    Object.keys(this.textContentDivs).map(textContentID => {
      const text = this.state.texts.find( x => x.id === parseInt(textContentID) ).text
      this.textContentDivs[textContentID].insertAdjacentHTML('beforeend', text)
    })
  }

  defineHeight = (map) => {
    return document.body.clientWidth < 768 ? `${map.height}px` : "auto%"
  }

  definePhotoWidth = (photo) => { return this.state.clientWidth >= 768 ? photo.css_width : 100}

  activateSecondContent = (doubleContentID) => {
    let doubleContentActives = this.state.doubleContentActives
    let initialState = doubleContentActives[`doubleContent-${doubleContentID}`]
    doubleContentActives = update(doubleContentActives, { [`doubleContent-${doubleContentID}`]: { $set: !initialState }})
    this.setState({ doubleContentActives: doubleContentActives })
  }

  render() {
    console.log("render")
    console.log(this.state.doubleContentActives)
    return (
      <div className="container article-container">
        <div className="row">
           <div className="col">

            <header ref={this.articleContent}>
              <h1 className="article-header">{this.state.title}</h1>
            </header>

            <div className="article-content">
              {this.state.articleElements.map(element => {

                if (element.class_name == "TextContent") {
                  return <div key={`text-${element.id}`}
                  ref={div => this.textContentDivs[element.id] = div} className="textContent"></div>
                }

                if (element.class_name == "Map") {
                  return <div key={`map-${element.id}`} id={`map-${element.id}`}
                  ref={div => this.mapDivs[element.id] = div}
                  className="googleMap" style={{ width: '100%', height: `${element.height}px` }}></div>
                }

                if (element.class_name == "Photo") {
                  return <div className="photo" key={`photo-${element.id}`}>
                    <img src={ photoHelpers.retrievePhotoSRC(element) } alt={ element.original_filename }
                    style={{width: `${this.definePhotoWidth(element)}%`}} />
                    { element.display_title && <p className="photo-title">{ element.original_filename }</p> }
                  </div>
                }

                if (element.class_name == "DoubleContent") {
                  return (
                      <div className={`doubleContent
                        ${this.state.doubleContentActives ? (this.state.doubleContentActives[`doubleContent-${element.id}`] ? "active" : "") : null}`}
                       key={`doubleContent-${element.id}`} style={{minHeight: `${element.height}px`}}>
                        <ShowSecondContentButton activateSecondContent={this.activateSecondContent} text={"Show second content"}
                          id={element.id} />
                      { element.associated_instances_mapping.map( (type, index) => {
                        if (type[0] === "TextContent") {
                          const text_content = element.text_contents[index] || element.text_contents[0]
                          return <div key={`text-${text_content.id}`}
                            className={`textContent ${text_content.position === 0 ? "firstContent" : "secondContent"}`}
                            ref={div => this.textContentDivs[text_content.id] = div} >
                          </div>
                        }
                        if (type[0] === "Map") {
                          const map = element.maps[index] || element.maps[0]
                          return <div key={`map-${map.id}`} id={`map-${map.id}`} ref={div => this.mapDivs[map.id] = div}
                            className={`googleMap ${map.position === 0 ? "firstContent" : "secondContent"}`}
                            style={{ height: `${this.defineHeight(map)}` }}>
                          </div>
                        }
                        if (type[0] === "Photo") {
                          const photo = element.photos[index] || element.photos[0]
                          return <div key={`photo-${photo.id}`}
                            className={`photo ${photo.position === 0 ? "firstContent" : "secondContent"}`}
                            style={{ backgroundImage: `url(${photo.url})`}}>
                              <img className="photoMobile"src={photo.url} alt=""/>

                          </div>
                        }
                      })}
                    </div>
                  )
                }
              })}
            </div>

          </div>
        </div>
      </div>
    )
  }
}

Article.propTypes = {
  match: PropTypes.object.isRequired
}

function mapStateToProps(state) {
  return {
    currentArticle: state.currentArticle
  }
}

export default withRouter(connect(mapStateToProps, null)(Article))
