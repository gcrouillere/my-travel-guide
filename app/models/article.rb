class Article < ApplicationRecord
  has_many :text_contents, -> { order('position ASC') }, dependent: :destroy
  has_many :maps, -> { order('position ASC') }, dependent: :destroy
  has_many :photos, -> { order('position ASC') }, dependent: :destroy
  has_many :double_contents, -> { order('position ASC') }, dependent: :destroy
  has_many :article_audience_selections
  has_many :audience_selections, through: :article_audience_selections
  belongs_to :user

  include AlgoliaSearch

  algoliasearch auto_index: false, auto_remove: false do
    attribute :title
    add_attribute :article_id
  end

  def elements_position_mapping
    self.as_json(include: {
        text_contents: { methods: :class_name },
        maps: { methods: :class_name },
        photos: { methods: :class_name },
        double_contents: { methods: :class_name },
      }).select {|k, v| k.match(/^maps$|^text_contents$|^photos$|^double_contents$/)}
      .select {|k, v| v != []}
      .map {|k, v| v.map {|e| [e["class_name"], e["id"], e["position"]] }}
      .flatten(1)
  end

  def article_id
    self.id
  end
end
