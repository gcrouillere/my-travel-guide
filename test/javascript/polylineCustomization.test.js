import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router'
import renderer from 'react-test-renderer';

import article from './__mocks__/fakeArticle'
import PolylineCustomization from '../../app/javascript/packs/articleForm/mapForm/polylineCustomization'
import MapComponent from '../../app/javascript/packs/articleForm/mapForm/mapComponent'
import Polyline from '../../app/javascript/packs/articleForm/mapForm/mapComponent/polyline'
import MapForm from '../../app/javascript/packs/articleForm/mapForm'

const map = article.maps[0]

const polyline = map.polylines[0]

configure({ adapter: new Adapter() });

jest.mock('../../app/javascript/utils/ajaxHelpers')

describe('PolylineCustomization tests', () => {

  it('renders component', () => {
    const PolylineCustomizationTree = renderer
      .create(<PolylineCustomization map={map} customizationOnGoing={false}/>)
      .toJSON()
      expect(PolylineCustomizationTree).toMatchSnapshot();
  })

  let preventDraggingOnOtherElements = jest.fn()

  let wrapper = mount(
    <MapForm map={map} preventDraggingOnOtherElements={preventDraggingOnOtherElements}>
      <PolylineCustomization map={map} customizationOnGoing={false}/>
    </MapForm>
    )

  it('reveals PolylineCustomization overlay on click', () => {
    wrapper.find(PolylineCustomization).instance().initPolylineCustomization({}, {}, polyline, {})
    wrapper.update()

    expect(wrapper.find(Polyline).length).toEqual(1)
    expect(wrapper.find(".polylineCustomization.active").length).toEqual(1)
  })

  it('displays distance on click', async () => {
    await wrapper.find(".polylineCustomization #radio-true").simulate('change')
    wrapper.update()

    expect(wrapper.find(Polyline).at(0).state().polyline.distance_displayed).toEqual(true)
  })

  it('adds a point on function call', async () => {
    let event = { latLng: {
      lat: jest.fn().mockImplementation(x => { return 2 }),
      lng: jest.fn().mockImplementation(x => { return 2 })
    }}
    let googlePolyline = new google.maps.Polyline
    wrapper.find(PolylineCustomization).setState({polyline: {id: 1}})
    await wrapper.find(PolylineCustomization).instance().continuePath(event, googlePolyline)

    expect(wrapper.find(PolylineCustomization).state().polylineMarkers.length).toEqual(3)
  })

  it('deletes a marker on function call', async () => {
    await wrapper.find(PolylineCustomization).instance().deletePointFromPath()

    expect(wrapper.find(PolylineCustomization).state().polylineMarkers.length).toEqual(2)
  })

  it('deletes polyline on function call', async () => {
    await wrapper.find(PolylineCustomization).instance().deletePolyline()

    expect(wrapper.find(MapComponent).state().polylines.length).toEqual(0)
  })
})
