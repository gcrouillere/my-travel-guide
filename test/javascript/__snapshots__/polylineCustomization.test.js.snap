// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolylineCustomization tests renders component 1`] = `
<div
  className="polylineCustomization "
  id="polylineCustomization-2"
>
  <div
    className="overflowContainer"
  >
    <button
      aria-label="Close"
      className="close"
      onClick={[Function]}
    >
      <span
        aria-hidden="true"
      >
        Ã—
      </span>
    </button>
    <h3>
      Path Customization:
    </h3>
    <div
      className="distanceDisplayRadios mapCustomizationBlock"
    >
      <p>
        Display path length ?
      </p>
      <div
        className="form-check form-check-inline"
      >
        <input
          checked={false}
          className="form-check-input markerLogoInput"
          disabled={false}
          id="radio-true"
          name="inlineRadioOptions"
          onChange={[Function]}
          type="radio"
          value={true}
        />
        <label
          className="form-check-label"
          htmlFor="radio-true"
        >
          Yes
        </label>
      </div>
      <div
        className="form-check form-check-inline"
      >
        <input
          checked={false}
          className="form-check-input markerLogoInput"
          disabled={false}
          id="radio-false"
          name="inlineRadioOptions"
          onChange={[Function]}
          type="radio"
          value={false}
        />
        <label
          className="form-check-label"
          htmlFor="radio-false"
        >
          No
        </label>
      </div>
    </div>
    <button
      className="btn btn-dark mapCustomizationBlock startPath"
      disabled={false}
      onClick={[Function]}
    >
      Continue Path
    </button>
    <button
      className="btn btn-dark mapCustomizationBlock startPath"
      disabled={false}
      onClick={[Function]}
    >
      Delete a Point from Path
    </button>
    <button
      className="btn btn-dark"
      disabled={false}
      onClick={[Function]}
    >
      Delete Path
    </button>
  </div>
</div>
`;
